snapshots/snapshots-processor/src/main/kotlin/com/emergetools/snapshots/processor/utils/PreviewFinderUtils.kt

```diff
@@ -36,9 +36,44 @@ fun List<KSAnnotated>.functionsWithMultiPreviewAnnotation(
         .map { function ->
           function to getUniqueSnapshotConfigsFromMultiPreviewAnnotation(
             annotations = multiPreviewAnnotationPreviewAnnotations,
-            previewFunction = function
+            previewFunction = function,
           )
         }
     }
-    .toMap()
+
+  // The same function declaration can show up multiple times, so ensure the values are merged together
+  val mergedConfigs = mutableMapOf<KSFunctionDeclaration, List<ComposePreviewSnapshotConfig>>()
+  uniqueSnapshotConfigs.forEach {
+    mergedConfigs.putOrAppend(it.first, it.second)
+  }
+
+  return mergedConfigs
+}
+
+fun Resolver.getMultiPreviewAnnotations(): List<KSClassDeclaration> {
+  // Find all symbols with annotations and map to the annotation class declarations
+  val annotationClassDecls = getAllFiles()
+    .flatMap { it.declarations }
+    .filter { it.annotations.count() > 0 }
+    .flatMap { symbol ->
+      symbol.annotations.mapNotNull { annotation ->
+        val annotationQN = annotation.annotationType.resolve().declaration.qualifiedName
+        annotationQN?.let { qualifiedName ->
+          getClassDeclarationByName(qualifiedName)
+        }
+      }
+    }
+    .filter { it.classKind == ClassKind.ANNOTATION_CLASS }
+    .toSet()
+
+  // Of the annotation classes we found, take those that themselves have a preview annotation.
+  // We can assume these are multi-preview annotations.
+  return annotationClassDecls
+    .filter {
+      it.annotations.any { annotation ->
```

Comment:
Will this get annotations that are transitively added to this annotation? Ex:

```
@Preview(...)
annotation class Class1

@Class1
annotation class Class2
```

Will it pick up Class2 as having a preview annotation? If not can we do this in a loop building up what annotations are eligible to make a preview?

Reply:
We had discussed v1 of this only supporting a single level of Multipreview nesting (and checked with clients that that was ok), so that means we're only intending to cover your first example here.

But, if it's easy enough to add, we might as well add support for all potential cases here.